src/Controller/Controller.php

```diff
@@ -221,8 +212,7 @@ class Controller implements EventListenerInterface, EventDispatcherInterface
      * @see \Cake\View\View
      */
     protected $_validViewOptions = [
-        'viewVars', 'autoLayout', 'helpers', 'view', 'layout', 'name', 'theme', 'layoutPath',
-        'viewPath', 'plugin', 'passedArgs'
+        'viewVars', 'helpers', 'name', 'layoutPath', 'viewPath', 'plugin', 'passedArgs'
```

Comment:
I was at a dilemma regarding layoutPath and viewPath. These properties can be set by request handler causing view object being created much early in the request even though view rendering might not be required. I am not too happy with that happening.